Backend/src/usecase/implementation/UpdateGameInteractor.ts

```diff
@@ -16,17 +17,27 @@ export class UpdateGameInteractor implements UpdateGameUseCase {
         this.gameB2DConverter = new GameB2DConverter();
     }
 
-    updateGame(gameBoundary: BoundaryGame): BoundaryGame {
+    /*updateGame(gameBoundary: BoundaryGame): BoundaryGame {
         const foundGame = this.getGameFromDatabase(gameBoundary.getId())
         this.tryUpdateGameInDatabase(gameBoundary, foundGame.getWord())
         const updatedGame = this.getGameFromDatabase(gameBoundary.getId())
         return this.gameD2BConverter.convert(updatedGame)
+    }*/
+
+    updateGame(game: Game): Game {
+        this.tryUpsertGameInDatabase(game)
+        const updatedGame = this.getGameFromDatabase(game.getId())
+        return updatedGame;
+    }
+
+    private tryUpsertGameInDatabase(game:Game){
```

Comment:
ne database, slėpk detales. gateway gali būti ir failas, ir kitas api, ir inmemory mapas
Šito metodo net nereikia, tu pirminiam "updateGame" metode tiesiog kviesk this.gamesGateway.upsertGame(game);